src/meta/processors/admin/Balancer.cpp

```diff
@@ -424,30 +466,74 @@ bool Balancer::balanceParts(BalanceID balanceId,
         return false;
       }
 
-      partsFrom.erase(it);
       if (std::find(partsTo.begin(), partsTo.end(), partId) != partsTo.end()) {
         LOG(ERROR) << "Part " << partId << " already existed in partsTo";
         return false;
       }
 
+      if (dependentOnGroup) {
+        auto& parts = relatedParts_[minPartsHost.first];
+        if (!checkZoneLegal(maxPartsHost.first, minPartsHost.first) &&
+            std::find(parts.begin(), parts.end(), partId) != parts.end()) {
+          LOG(INFO) << "Zone have exist part: " << partId;
+          continue;
+        }
+      }
+
+      partsFrom.erase(it);
       partsTo.emplace_back(partId);
       tasks.emplace_back(
           balanceId, spaceId, partId, maxPartsHost.first, minPartsHost.first, kv_, client_);
       noAction = false;
     }
+
     if (noAction) {
       LOG(INFO) << "Here is no action";
       break;
     }
     sortedHosts = sortedHostsByParts(confirmedHostParts);
     maxPartsHost = sortedHosts.back();
     minPartsHost = sortedHosts.front();
+    if (innerBalance_) {
+      LOG(INFO) << "maxPartsHost.first " << maxPartsHost.first << " minPartsHost.first "
+                << minPartsHost.first;
```

Comment:
don't output the code block for log. WDYT?

Reply:
ok